crates/loader/src/local/mod.rs

```diff
@@ -405,20 +402,40 @@ impl ComponentDigest {
 fn resolve_partials(
     partially_parsed: RawAppManifestAnyVersionPartial,
 ) -> Result<RawAppManifestAnyVersion> {
-    let RawAppManifestAnyVersionPartial::V1(manifest) = partially_parsed;
+    // TODO: refactor this to be cleaner
+    let (version, manifest, is_old) = match partially_parsed {
+        RawAppManifestAnyVersionPartial::V1Old {
+            spin_version: v,
+            manifest,
+        } => (v, manifest, true),
+        RawAppManifestAnyVersionPartial::V1New {
+            spin_manifest_version: v,
+            manifest,
+        } => (v, manifest, false),
+    };
 
     let app_trigger = &manifest.info.trigger;
     let components = manifest
         .components
         .into_iter()
         .map(|c| resolve_partial_component(app_trigger, c))
         .collect::<Result<_>>()?;
-
-    Ok(RawAppManifestAnyVersion::V1(RawAppManifest {
+    let full_manifest = RawAppManifest {
         info: manifest.info,
         components,
         variables: manifest.variables,
-    }))
+    };
+
+    Ok(match is_old {
```

Comment:
I don't believe you need to preserve oldness and newness here.  I think it's safe to always emit it as new - nothing cares about oldness or newness, just about the content, which remains identical.

Reply:
@fibonacci1729 and I were wondering about this. Sounds good! Thanks

Reply:
Cleaned this up since we are only concerned with maintaining the `manifest`